conf/i18n/extract/translationextractor.js

```diff
@@ -0,0 +1,108 @@
+const { GettextExtractor } = require('gettext-extractor');
+const Handlebars = require('handlebars');
+const fsExtra = require('fs-extra');
+const { fromMustacheStatementNode } = require('../translationplaceholderutils');
+const { TRANSLATION_FLAGGER_REGEX } = require('../constants');
+const TranslateCallParser = require('../translatecallparser');
+
+/**
+ * TranslationExtractor extracts translations to a gettext style file.
+ */
+class TranslationExtractor {
+  constructor () {
+    this._extractor = new GettextExtractor();
+    this._translateHelpers = ['translate'];
+  }
+
+  /**
+   * Extracts messages from a given hbs template.
+   * 
+   * @param {string} template 
+   * @param {string} filepath 
+   */
+  extractFromHBS (template, filepath) {
+    const tree = Handlebars.parseWithoutProcessing(template);
+    const visitor = new Handlebars.Visitor();
+    visitor.MustacheStatement = statement =>
+      this._extractFromMustacheNode(statement, filepath);
+    visitor.accept(tree);
+  }
+
+  /**
+   * Extracts messages from javascript code.
+   * 
+   * @param {string} code
+   * @param {string} filepath
+   */
+  extractFromJS (code, filepath) {
+    const matches = [...code.matchAll(TRANSLATION_FLAGGER_REGEX)];
+    for (const match of matches) {
+      this._extractFromJSCall(match, filepath);
+    }
+  }
+
+  /**
+   * Returns the extracted messages as a string.
+   *
+   * @returns {Array<Object>}
+   */
+  getPotString () {
```

Comment:
Is this used?

Reply:
nope was using this just for testing! will remove